src/Guards/TokenGuard.php

```diff
@@ -65,56 +83,79 @@ class TokenGuard
      * @param  \Laravel\Passport\TokenRepository  $tokens
      * @param  \Laravel\Passport\ClientRepository  $clients
      * @param  \Illuminate\Contracts\Encryption\Encrypter  $encrypter
+     * @param  \Illuminate\Http\Request  $request
      * @return void
      */
     public function __construct(
         ResourceServer $server,
         PassportUserProvider $provider,
         TokenRepository $tokens,
         ClientRepository $clients,
-        Encrypter $encrypter
+        Encrypter $encrypter,
+        Request $request
     ) {
         $this->server = $server;
         $this->tokens = $tokens;
         $this->clients = $clients;
         $this->provider = $provider;
         $this->encrypter = $encrypter;
+        $this->request = $request;
     }
 
     /**
      * Get the user for the incoming request.
      *
-     * @param  \Illuminate\Http\Request  $request
      * @return mixed
      */
-    public function user(Request $request)
+    public function user()
     {
-        if ($request->bearerToken()) {
-            return $this->authenticateViaBearerToken($request);
-        } elseif ($request->cookie(Passport::cookie())) {
-            return $this->authenticateViaCookie($request);
+        if (! is_null($this->user)) {
+            return $this->user;
+        }
+
+        if ($this->request->bearerToken()) {
+            return $this->user = $this->authenticateViaBearerToken($this->request);
+        } elseif ($this->request->cookie(Passport::cookie())) {
+            return $this->user = $this->authenticateViaCookie($this->request);
         }
     }
 
+    /**
+     * Validate a user's credentials.
+     *
+     * @param  array  $credentials
+     * @return bool
+     */
+    public function validate(array $credentials = [])
+    {
+        return ! is_null((new static(
+            $this->server,
+            $this->provider,
+            $this->tokens,
+            $this->clients,
+            $this->encrypter,
+            $credentials['request'],
+        ))->user());
+    }
+
     /**
      * Get the client for the incoming request.
      *
-     * @param  \Illuminate\Http\Request  $request
      * @return mixed
      */
-    public function client(Request $request)
+    public function client()
     {
-        if ($request->bearerToken()) {
-            if (! $psr = $this->getPsrRequestViaBearerToken($request)) {
+        if ($this->request->bearerToken()) {
```

Comment:
You probably want to return the already resolved client here early like the user method does.

Reply:
Oops, fixing that 👍